url3f8d6440dd_huiwq1990_CognativeModel_tgz-pJ8-clarion_samples_HelloWorldJ8/src/clarion/samples/XOR.java

```diff
@@ -0,0 +1,343 @@
+		//acsbp.MOMENTUM = .5;
+		//acsbp.LEARNING_RATE = .2;
+		
+		//Add the neural net to the bottom level of the ACS.
+		acs.addBLModule(acsbp);
+	}
+
+	/**
+	 * This method initializes the sensory input space and the combined input space.
+	 */
+	public void initializeSensoryInformationSpace() {
+		//BEGIN initialization of the sensory input space.
+		//Initialize a new dimension-value collection.
+		SensoryInputSpace = new DimensionValueCollection ();
+		//Add a new dimension.
+		SensoryInputSpace.put("Boolean1",new Dimension ("Boolean1"));
+		//Add new values to the dimension.
+		SensoryInputSpace.get("Boolean1").put("True", new Value("True"));
+		SensoryInputSpace.get("Boolean1").put("False", new Value("False"));
+		
+		//Add a new dimension.
+		SensoryInputSpace.put("Boolean2",new Dimension ("Boolean2"));
+		//Add new values to the dimension.
+		SensoryInputSpace.get("Boolean2").put("True", new Value("True"));
+		SensoryInputSpace.get("Boolean2").put("False", new Value("False"));
+		
+		/*Initialize the combined input space with a new dimension-value that contains the
+		 * sensory input space.*/
+		CombinedInputSpace = new DimensionValueCollection (SensoryInputSpace);
+		//END initialization of the sensory input space.
+	}
+
+	/**
+	 * This method reports the results of each agent during the simulation.
+	 */
+	public void reportResults() {
+		Accuracy = CorrectCounter/NumberTrials;
+		CorrectCounter = 0;
+		double percentage = Accuracy*100.0;
+		BigDecimal bd = new BigDecimal( Double.toString(percentage) );
+		bd = bd.setScale( DecimalPlace, BigDecimal.ROUND_HALF_UP );
+
+		if (CurrentTest == NumberTestRuns){
+			System.out.println(" |"+bd.doubleValue()+"|- run complete");
+			if (useVerboseLogging){
+				System.out.println("The agent learned the following rules through RER:");
+				for(AbstractRule r : Agent.getACS().getRERRules())
+				{
+					//Report the rules the agent learned using RER.
+					System.out.println(r.toString());		
+				}
+			}
+		}
+		else
+			System.out.print(" |"+bd.doubleValue()+"| ");
+			
+	}
+		
+	/**
+	 * This method reports the final results after all agents have been run.
+	 */
+	public void reportFinalResults(){
+		System.out.println("All tests and trials complete.");
+		double percentAccuracy = (AvgAccuracyNumerator/NumberAgents)*100.0;
+		CorrectCounter = 0;
+		BigDecimal bd = new BigDecimal( Double.toString(percentAccuracy) );
+		bd = bd.setScale( DecimalPlace, BigDecimal.ROUND_HALF_UP );
+		
+		System.out.println("Number of Trials per Test: "+ NumberTrials);
+		System.out.println("Number of Tests per Run: "+NumberTestRuns);
+		System.out.println("Number of Trials per Run:"+ NumberTestRuns*NumberTrials);
+		System.out.println("Number of Agents: "+ NumberAgents);
+		System.out.println("Avg (accuracy after all tests) over all agents: "+ bd.doubleValue());
+	}
+	
+	/**
+	 * This method runs an agent through the XOR task.
+	 */
+	public void run(CLARION Agent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.